pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,205 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+type IPAMClaimsManager struct {
+	client kubecli.KubevirtClient
+}
+
+func NewIPAMClaimsManager(client kubecli.KubevirtClient) *IPAMClaimsManager {
+	return &IPAMClaimsManager{
+		client: client,
+	}
+}
+
+type IPAMClaimParams struct {
+	ClaimName   string
+	NetworkName string
+}
+
+func (m *IPAMClaimsManager) CreateIPAMClaims(namespace string, vmiName string, interfaces []virtv1.Interface, networks []virtv1.Network, ownerRef *v1.OwnerReference) error {
+	nonAbsentNetworks := filterNonAbsentNetworks(interfaces, networks)
```

Comment:
I think the default networks should be filtered out here as well.

Reply:
I meant I think both of the filtering should be done in the same place. What is the reason to split it?

Reply:
callers of GetNetworkToIPAMClaimParams:
CalculateInterfacesAndNetworksForMultusAnnotationUpdate - calculates network list for update and then call GetNetworkToIPAMClaimParams
CreateIPAMClaims - filterNonAbsentNetworks and then calls GetNetworkToIPAMClaimParams

kept at GetNetworkToIPAMClaimParams the common filtering, and on each of the callers does what it needs beside
otherwise we will need to add filtering in each of the callers of GetNetworkToIPAMClaimParams

Reply:
Done